packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/classes.ts

```diff
@@ -274,6 +274,59 @@ export abstract class HandlerFrameworkClass extends ClassType {
           stmt.ret(expr.directCode('this.getOrCreateProvider(scope, uniqueid, props).serviceToken')),
         );
 
+        const idStatement = stmt.constVar(expr.ident('id'), expr.directCode('`${uniqueid}CustomResourceProvider`'));
+        const stackFromScopeStatement = stmt.constVar(expr.ident('stack'), expr.directCode('Stack.of(scope)'));
+        const logContextKeyStatement = stmt.constVar(expr.ident('key'), expr.directCode('`${uniqueid}CustomResourceLogGroup`'));
+        const getProviderMethod = this.addMethod({
+          name: 'getProvider',
+          static: true,
+          returnType: this.type,
+          docs: {
+            summary: 'Returns the stack-level singleton provider or undefined',
+          },
+        });
+        getProviderMethod.addParameter({
+          name: 'scope',
+          type: CONSTRUCTS_MODULE.Construct,
+        });
+        getProviderMethod.addParameter({
+          name: 'uniqueid',
+          type: Type.STRING,
+        });
+        getProviderMethod.addBody(
+          idStatement,
+          stackFromScopeStatement,
+          stmt.ret(expr.directCode(`stack.node.tryFindChild(id) as ${this.type}`)),
+        );
+
+        const useLogGroupMethod = this.addMethod({
+          name: 'useLogGroup',
+          static: true,
+          docs: {
+            summary: 'Set the log group to be used by the singleton provider',
+          },
+        });
```

Comment:
Technically you're adding `useLogGroup` functionality to all custom resource handlers not only the s3 ones right? But the functionality is only exposed to the s3 auto delete handler. It would be nice to expose `useLogGroup` to other handlers as I believe this issue doesn't only happen on s3 auto delete but also on all custom resource handlers.